R/plot.data_table_threads_benchmark.R

```diff
@@ -23,39 +23,56 @@
 
 plot.data_table_threads_benchmark <- function(x, ...)
 {
-  df <- x
-  rownames(df) <- NULL
-  df$speedup <- df$medianTime[df$threadCount == 1] / df$medianTime
-
-  setDT(df)
-  maxSpeedup <- df[, .(threadCount = threadCount[which.max(speedup)], speedup = max(speedup)), by = expr]
-  idealSpeedup <- seq(1, getDTthreads())
-  idealSpeedupData <- data.frame(threadCount = 1:getDTthreads(), speedup = idealSpeedup)
-  subOptimalSpeedupData <- data.frame(threadCount = seq(1, getDTthreads(), length.out = getDTthreads()), speedup = seq(1, getDTthreads()/2, length.out = getDTthreads()))
-
-  closestPoints <- data.frame()
-  for(i in unique(df$expr)) {
-    dfSubset <- df[df$expr == i, ]
-    suboptimalSubset <- subOptimalSpeedupData[subOptimalSpeedupData$threadCount %in% dfSubset$threadCount, ]
-    closestPoint <- dfSubset[which.max(dfSubset$speedup - suboptimalSubset$speedup), ]
-    closestPoints <- rbind(closestPoints, closestPoint)
-  }
-
-  ggplot(df, aes(x = threadCount, y = speedup)) +
-    geom_line(aes(linetype = "Measured")) +
-    geom_line(data = idealSpeedupData, aes(x = threadCount, y = speedup, linetype = "Ideal"), color = "black") +
-    geom_line(data = subOptimalSpeedupData, aes(x = threadCount, y = speedup, linetype = "Sub-optimal"), color = "black") +
-    geom_point(data = closestPoints, aes(x = threadCount, y = speedup, shape = "Recommended"), color = "black", size = 2) +
-    geom_point(data = maxSpeedup, aes(x = threadCount, y = speedup, shape = "Best performing"), color = "red", size = 2) +
-    geom_text(data = closestPoints, aes(label = threadCount), vjust = -0.5, size = 4, na.rm = TRUE) +
-    geom_text(data = maxSpeedup, aes(label = threadCount), vjust = -0.5, size = 4, na.rm = TRUE) +
-    geom_ribbon(aes(ymin = speedup - 0.3, ymax = speedup + 0.3), alpha = 0.5) +
+  x[, `:=`(
+    speedup = medianTime[threadCount == 1] / medianTime,
+    type = "Measured"
+  ), by = expr]
+
+  maxSpeedup <- x[, .(threadCount = threadCount[which.max(speedup)], 
+                      speedup = max(speedup)), by = expr]
+
+  idealSpeedup <- x[, .(threadCount = 1:getDTthreads(),
+                        speedup = seq(1, getDTthreads()),
+                        type = "Ideal",
+                        medianTime = NA), by = expr]
+
+  subOptimalSpeedup <- x[, .(threadCount = seq(1, getDTthreads(), length.out = getDTthreads()),
+                             speedup = seq(1, getDTthreads() / 2, length.out = getDTthreads()),
+                             type = "Sub-optimal",
+                             medianTime = NA), by = expr]
```

Comment:
please combine this logic instead of repeating it for both values of `type`
your code should do 
```r
speedup <- something[, .(...), by=.(expr, type)]
```